src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -1913,6 +1919,31 @@ public void cannotPostSynchronousBatchWithoutStatus() {
     }
   }
 
+  @Test
+  @Parameters(method = "getAllowedItemStatuses")
+  public void canCreateItemWithAllAllowedStatuses(Status.Name status) throws Exception {
+    final UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    final ItemRequestBuilder itemToCreate = new ItemRequestBuilder()
+      .forHolding(holdingsRecordId)
+      .withMaterialType(journalMaterialTypeId)
+      .withPermanentLoanType(canCirculateLoanTypeId)
+      .withStatus(status.value());
+
+    final IndividualResource createdItem = itemsClient.create(itemToCreate);
+    assertThat(createdItem.getJson().getJsonObject("status").getString("name"),
+      is(status.value()));
+
+    JsonObject itemInStorage = itemsClient.getById(createdItem.getId()).getJson();
+    assertThat(itemInStorage.getJsonObject("status").getString("name"),
+      is(status.value()));
+  }
+
+  @SuppressWarnings("unused")
+  private Status.Name[] getAllowedItemStatuses() {
+    return Status.Name.values();
```

Comment:
I think we need to be cautious with this kind of testing. It demonstrates that the API respects the generated code, rather than the generated coding produces the expected behaviour.

Reply:
Do you suggest to hardcode all the allowed statuses instead?

Reply:
I suggest we either hard code the values or we use the source of truth which is the schema (as I think you have done in the mod-inventory change)?

Reply:
Test uses schema as source of allowed statuses now.